faucet/src/main.rs

```diff
@@ -27,6 +31,33 @@ struct Opts {
     /// Settings specific to the faucet client.
     #[clap(flatten)]
     faucet: FaucetConfig,
+
+    /// Allowance config
+    #[clap(long, default_value = "./faucet-allowance-config.json")]
+    allowance_config: PathBuf,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct AllowanceAmount {
+    ticker: String,
+    amount: u128,
+}
+
+type AllowanceTuple = (CurrencyId, u128);
+type Allowance = Vec<AllowanceTuple>;
+impl From<&AllowanceAmount> for AllowanceTuple {
+    fn from(allowance_amount: &AllowanceAmount) -> Self {
+        (
+            CurrencyId::try_from_symbol(allowance_amount.ticker.clone()).expect("Failed to parse input ticker"),
+            allowance_amount.amount,
+        )
+    }
+}
```

Comment:
I think this is unnecessary, let's just use one struct for this `AllowanceAmount`.